mechanics/physicell/src/vtk_agents_serializer.cpp

```diff
@@ -0,0 +1,352 @@
+		relative_maximum_adhesion_distance_array->SetValue(static_cast<vtkIdType>(i),
+														   data.mechanics.relative_maximum_adhesion_distance[i]);
+		maximum_number_of_attachments_array->SetValue(
+			static_cast<vtkIdType>(i), static_cast<real_t>(data.mechanics.maximum_number_of_attachments[i]));
+		attachment_elastic_constant_array->SetValue(static_cast<vtkIdType>(i),
+													data.mechanics.attachment_elastic_constant[i]);
+		attachment_rate_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.attachment_rate[i]);
+		detachment_rate_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.detachment_rate[i]);
+
+		is_motile_array->SetValue(static_cast<vtkIdType>(i), static_cast<real_t>(data.motility.is_motile[i]));
+		persistence_time_array->SetValue(static_cast<vtkIdType>(i), data.motility.persistence_time[i]);
+		migration_speed_array->SetValue(static_cast<vtkIdType>(i), data.motility.migration_speed[i]);
+		migration_bias_array->SetValue(static_cast<vtkIdType>(i), data.motility.migration_bias[i]);
+		restrict_to_2d_array->SetValue(static_cast<vtkIdType>(i), static_cast<real_t>(data.motility.restrict_to_2d[i]));
+		chemotaxis_index_array->SetValue(static_cast<vtkIdType>(i),
+										 static_cast<real_t>(data.motility.chemotaxis_index[i]));
+		chemotaxis_direction_array->SetValue(static_cast<vtkIdType>(i),
+											 static_cast<real_t>(data.motility.chemotaxis_direction[i]));
+		simple_pressure_array->SetValue(static_cast<vtkIdType>(i), data.state.simple_pressure[i]);
+		cell_definition_index_array->SetValue(static_cast<vtkIdType>(i),
+											  static_cast<real_t>(data.state.agent_type_index[i]));
+		is_movable_array->SetValue(static_cast<vtkIdType>(i), static_cast<real_t>(data.state.is_movable[i]));
+
+		std::array<double, 3> velocity = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> previous_velocity = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> migration_bias_dir = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> motility_vector = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> orientation = { 0.0, 0.0, 0.0 };
+
+		for (index_t d = 0; d < dims && d < 3; ++d)
+		{
+			const index_t offset = i * dims + d;
+			velocity[d] = data.velocity[offset];
+			previous_velocity[d] = data.previous_velocity[offset];
+			migration_bias_dir[d] = data.motility.migration_bias_direction[offset];
+			motility_vector[d] = data.motility.motility_vector[offset];
+			orientation[d] = data.state.orientation[offset];
+		}
+
+		velocity_array->SetTuple(static_cast<vtkIdType>(i), velocity.data());
+		previous_velocity_array->SetTuple(static_cast<vtkIdType>(i), previous_velocity.data());
+		migration_bias_direction_array->SetTuple(static_cast<vtkIdType>(i), migration_bias_dir.data());
+		motility_vector_array->SetTuple(static_cast<vtkIdType>(i), motility_vector.data());
+		orientation_array->SetTuple(static_cast<vtkIdType>(i), orientation.data());
```

Comment:
is this even needed? cant you just do `velocity_array->SetTuple(static_cast<vtkIdType>(i), data.velocity.data() + i * dims);` ...